modules/i3s/src/lib/parsers/parse-i3s-attribute.ts

```diff
@@ -67,17 +67,21 @@ function parseFloatAttribute(arrayBuffer) {
 function parseStringsAttribute(arrayBuffer: ArrayBuffer): string[] {
   const dataOffset = 8;
   const bytesPerStringSize = 4;
-  const stringsCount = new Uint32Array(arrayBuffer, 0, bytesPerStringSize)[0];
-  const stringSizes = new Uint32Array(arrayBuffer, dataOffset, stringsCount);
-
   const stringsArray: string[] = [];
-  let stringOffset = dataOffset + stringsCount * bytesPerStringSize;
 
-  for (const stringByteSize of stringSizes) {
-    const textDecoder = new TextDecoder('utf-8');
-    const stringAttribute = new Uint8Array(arrayBuffer, stringOffset, stringByteSize);
-    stringsArray.push(textDecoder.decode(stringAttribute));
-    stringOffset += stringByteSize;
+  try {
+    const stringsCount = new Uint32Array(arrayBuffer, 0, bytesPerStringSize)[0];
+    const stringSizes = new Uint32Array(arrayBuffer, dataOffset, stringsCount);
+    let stringOffset = dataOffset + stringsCount * bytesPerStringSize;
```

Comment:
Do you just want to use the `padToNBytes` function?

FWIW I still don't understand the problem, why we need to pad `Uint8Arrays` to 4 bytes length.

Reply:
I think I don't need `padToNBytes` function just because I easily calculate it in one line only once.
`let stringOffset = dataOffset + stringsCount * bytesPerStringSize;`
The problem is not related to Uint8Array. It is related to Uint32Array.

Reply:
And I solved multiple of 4 issue with Uint32Array using DataView